pkg/osm/osminit.go

```diff
@@ -0,0 +1,48 @@
+package osm
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+	"github.com/openservicemesh/osm-health/pkg/kubernetes/podhelper"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*NoBadOsmInitLogsCheck)(nil)
+
+// NoBadOsmInitLogsCheck implements common.Runnable
+type NoBadOsmInitLogsCheck struct {
+	client kubernetes.Interface
+	pod    *corev1.Pod
+}
+
+// HasNoBadOsmInitLogsCheck checks whether the osm-init container of the pod has bad (fatal/error/warning/fail) log messages
+func HasNoBadOsmInitLogsCheck(client kubernetes.Interface, pod *corev1.Pod) NoBadOsmInitLogsCheck {
+	return NoBadOsmInitLogsCheck{
+		client: client,
+		pod:    pod,
+	}
+}
+
+// Info implements common.Runnable
+func (check NoBadOsmInitLogsCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s has bad (fatal/error/warning/fail) logs in osm-init container", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check NoBadOsmInitLogsCheck) Run() error {
+	return podhelper.HasNoBadLogs(check.client, check.pod, "osm-init")
```

Comment:
```suggestion
	return podhelper.HasNoBadLogs(check.client, check.pod, constants.InitContainerName)
```
(import "github.com/openservicemesh/osm/pkg/constants")